src/zcl_abapgit_repo.clas.abap

```diff
@@ -445,6 +463,7 @@ CLASS ZCL_ABAPGIT_REPO IMPLEMENTATION.
   METHOD refresh.
 
     mv_do_local_refresh = abap_true.
+    reset_remote( ).
```

Comment:
if applying the above suggestion this can be left out? Refresh will refresh the local objects(as for online repos), the user can choose to update the remote(git remote or zip remote) at any time, which will trigger new status calculation?

Reply:
I had doubts adding it. It is a bit cluttered indeed. Here is what I think: 
- there is `reset_remote` and `reset_status` - they really go together
- `reset_remote` is redefined in online, but just to clear `mv_initialized` in addition
- `mv_initialized` is set in `refresh` (in online) so actually symbolizes the same as `has_remote` ...
- ... however it is cleared in multiple places (set_url, set_branch ...)

What if
- combine `reset_remote` and `reset_status` into one `clear_remote` method in the base class (or a different name if there is a better option)
- move `mv_initialized` to base, clear it in the above `clear_remote`. And replace all `mv_initialized = abap_false` with `clear_remote` so that it is not cleared directly.
- instead of `has_remote` add `is_initialized`

Does it make sense ?